compiler/rustc_codegen_ssa/src/traits/consts.rs

```diff
@@ -5,7 +5,13 @@ use rustc_target::abi;
 pub trait ConstMethods<'tcx>: BackendTypes {
     // Constant constructors
     fn const_null(&self, t: Self::Type) -> Self::Value;
+    /// Generate an uninitialized value (matching uninitialized memory in MIR).
+    /// Whether memory is initialized or not is tracked byte-for-byte.
```

Comment:
I don't know whether it matters, but I think it might be bit-for-bit in LLVM?  The way I understand langref, `(undef << 1) & 1` in LLVM is definitely `0`, though in Rust I think we'd still treat it as `uninit`.

Reply:
Not sure if that's true for all backends though. Ultimately what matters is MIR semantics; it is then the backend's responsibility to find an appropriate translation.